products/dev_alternate_products.md

```diff
@@ -2,65 +2,96 @@
+Modifications are needed to the skeleton project's `build.gradle` and `plugin.xml` files, as described below, and on the individual product pages in Part VIII.
+The `build.gradle` file is modified to specify the target product, which determines the APIs available during development.
+The `plugin.xml` file is modified to declare the plugin's dependency on modules or libraries.
+
+## Configuring Build.Gradle to Target Products other than IntelliJ IDEA 
+The best practice is to use the IntelliJ IDEA Gradle plugin `intellij.type` [attribute](https://github.com/JetBrains/gradle-intellij-plugin/blob/master/README.md#intellij-platform-properties) to specify the target product.
+For example, `PY` for PyCharm professional.
+Configuration using an `intellij.type` attribute is explained in the [Product-Specific Attribute](#configuring-plugin-projects-using-a-product-specific-attribute) section below.
+
+However, not all products have an `intellij.type` attribute defined by the IntelliJ IDEA Gradle plugin.
+For example, PhpStorm.
+If the target product does not have an `intellij.type` attribute defined, then the best approach is to configure the project by using the [IntelliJ IDEA Attribute](#configuring-buildgradle-using-the-intellij-idea-product-attribute). 
+
+### Configuring Plugin Projects using a Product-Specific Attribute
+If the IntelliJ IDEA Gradle plugin supports a target product directly, there will be an `intellij.type` [attribute](https://github.com/JetBrains/gradle-intellij-plugin/blob/master/README.md#intellij-platform-properties) defined.
+Specifying the target as a product-specific `intellij.type` attribute has two advantages:
+* The APIs available to the plugin will be limited to only what is defined in the target product.
+  (Unless additional plugin dependencies are specified.)
+* The default [Development Instance](/basics/ide_development_instance.md) for running the plugin will be the target product.  
+
+A `build.gradle` snippet setting a plugin project to target PyCharm is shown below.
+The IntelliJ IDEA Gradle plugin will fetch the matching build of PyCharm Professional to define the APIs available, and use that build of PyCharm (and associated JetBrains runtime) as the Development Instance.
+No additional product-specific configuration needs to be set in `build.gradle`:
+```groovy
+    intellij {
+      version '2019.2.3'
+      type 'PY'
+    }
+```
+
+### Configuring Plugin Projects using the IntelliJ IDEA Product Attribute
+If the IntelliJ IDEA Gradle plugin does not directly support an IntelliJ Platform-based product, the `build.gradle` file can still be configured to target the desired product.
+In this case, the `build.gradle` file is configured to use IntelliJ IDEA as the basis for the available APIs.
```

Comment:
add "Community Edition"

Reply:
To be consistent with line 94, I changed line 53 to:
"In this case, the `build.gradle` file is configured to use IntelliJ IDEA (Community or Ultimate Edition) as the basis for the available APIs."